src/server_manager/model/account_manager.ts

```diff
@@ -0,0 +1,113 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import {LocalStorageRepository} from '../infrastructure/repository';
+
+import {Account, AccountId, DigitalOceanCredentials} from './account';
+import {CloudProviderId} from './cloud';
+import {DigitalOceanAccount} from '../web_app/digitalocean_app/digitalocean_account';
+import {DigitalOceanConnectAccountApp} from '../web_app/digitalocean_app/connect_account_app';
+
+export const ACCOUNT_MANAGER_KEY_EXTRACTOR = (entry: PersistedAccount) => entry.id;
+export const ACCOUNT_MANAGER_KEY_COMPARATOR = (k1: AccountId, k2: AccountId) =>
+    k1.cloudProviderId === k2.cloudProviderId &&
+    k1.cloudSpecificId === k2.cloudSpecificId;
+
+// TODO: Add migration from lastDOToken
+export const LEGACY_DIGITALOCEAN_ACCOUNT_ID: AccountId = {
+  cloudSpecificId: '_LEGACY_DIGITALOCEAN_ACCOUNT_ID_',
+  cloudProviderId: CloudProviderId.DigitalOcean,
+};
+
+/**
+ * A rich domain model that provides functionality to connect and manage cloud
+ * provider accounts.
+ */
+export class AccountManager {
+  // tslint:disable-next-line:no-any
+  private accountFactories: Map<CloudProviderId, AccountFactory<any>> = new Map();
+
+  constructor(private storageRepository: LocalStorageRepository<PersistedAccount, AccountId>) {}
+
+  initializeCloudProviders(digitalOceanConnectAccountApp: DigitalOceanConnectAccountApp) {
+    this.register(CloudProviderId.DigitalOcean, digitalOceanConnectAccountApp);
+  }
+
+  /** Loads existing DigitalOcean account credentials. */
+  async loadDigitalOceanAccount(): Promise<DigitalOceanAccount|undefined> {
+    return this.find(LEGACY_DIGITALOCEAN_ACCOUNT_ID) as Promise<DigitalOceanAccount|undefined>;
+  }
+
+  /**
+   * Connects to a DigitalOcean account.
+   *
+   * @param credentials - DigitalOcean account credentials.
+   */
+  async connectDigitalOceanAccount(credentials: DigitalOceanCredentials): Promise<DigitalOceanAccount> {
+    const persistedAccount = {
+      id: LEGACY_DIGITALOCEAN_ACCOUNT_ID,
+      credentials: credentials as unknown as object,
+    };
+    const account = await this.accountFactories.get(CloudProviderId.DigitalOcean).constructAccount(persistedAccount);
+    this.add(account);
+    return account;
+  }
+
+  /** Lists all connected accounts. */
+  listAccounts(): Promise<Account[]> {
+    const persistedAccounts = this.storageRepository.list();
+    return Promise.all(persistedAccounts.map((entry) => {
+      return this.loadAccount(entry);
+    }));
+  }
+
+  // TODO: Don't expose this
+  remove<T extends Account>(id: AccountId): void {
+    this.storageRepository.remove(id);
+  }
+
+  private async find(id: AccountId): Promise<Account|undefined> {
```

Comment:
`getAccount()` or `get()` would be clearer. It's not really a search, but a lookup. Get is consistent with the rest of the code.

Reply:
Thanks, will use `get()` for similar lookups going forward.

I removed this because it's no longer necessary with the new Cloud concept.